lix/packages/client/src/repoState.ts

```diff
@@ -103,41 +230,68 @@ export async function repoState(
 			throw new Error("fs provider does not support placeholders")
 		}
 		console.info("Using lix for cloning repo")
-
+		// FIXME: symlink = false in git/config!???
 		await isoGit
 			.clone({
-				fs: withProxy({ nodeishFs: rawFs, verbose: debug, description: "clone" }),
+				fs: rawFs,
 				http: makeHttpClient({
 					debug,
 					description: "clone",
-
-					onReq: ({ url, body }: { url: string; body: any }) => {
-						return optimizedRefsReq({ url, body, addRef: state.branchName })
-					},
-
-					onRes: optimizedRefsRes,
+					onReq: experimentalFeatures.lazyClone
+						? optimizeReq.bind(null, {
+								noBlobs: true,
+								addRefs: [state.branchName || "HEAD"],
+						  })
+						: undefined,
+					onRes: experimentalFeatures.lazyClone ? optimizeRes : undefined,
 				}),
 				dir,
 				cache,
 				corsProxy: gitProxyUrl,
 				url: gitUrl,
-				singleBranch: true,
+				singleBranch: false, // if we clone with single branch true we will not get the defatult branch set in isogit
 				noCheckout: experimentalFeatures.lazyClone,
 				ref: state.branchName,
 
 				// TODO: use only first and last commit in lazy clone? (we need first commit for repo id)
 				depth: 1,
 				noTags: true,
 			})
-			.then(() => {
+			.then(async () => {
 				if (!experimentalFeatures.lazyClone) {
 					return
 				}
-				return checkOutPlaceholders(ctx, {
-					branchName: state.branchName,
-					checkedOut: state.checkedOut,
-					sparseFilter: state.sparseFilter,
-				} as RepoState)
+				// console.log(
+				// 	"head",
+				// 	await isoGit
+				// 		.resolveRef({ fs: rawFs, dir: "/", ref: "HEAD", depth: 1 })
+				// 		.catch(console.error)
+				// )
+				// console.log(
+				// 	"base",
+				// 	await isoGit
+				// 		.resolveRef({ fs: rawFs, dir: "/", ref: "refs/remotes/origin/HEAD", depth: 1 })
+				// 		.catch(console.error)
+				// )
+				// console.log(
+				// 	"config",
+				// 	await isoGit.getConfig({ fs: rawFs, dir: "/", path: "." }).catch(console.error)
+				// )
+
+				const { gitignoreFiles } = await checkOutPlaceholders(
```

Comment:
We only load gitignoreFiles on fresh clone atm. i guess we need to also fetch them on pull